### PR TITLE
W-15213188 ext auth 1 7 gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 1]
```diff
@@ -16,11 +16,19 @@ endif::[]
...
+The External Authorization policy forwards incoming client HTTP requests to an external gRPC of HTTP authorization service to determine if the client HTTP request is authorized to access the upstream service. If the request is authorized, Flex Gateway allows the request to pass to the upstream service. For unauthorized accesses, Flex Gateway returns a `403` forbidden request error. 
 
-To forward request headers, such as an authorization token header, to the authorization service, use the *Allowed Headers* parameter to configure which headers from the client HTTP requests to send to the authorization service.
+To forward request headers, such as an authorization token header, to the authorization service, use the *Allowed Headers* parameter to configure which headers from the client HTTP requests to send to the authorization service. For policies with an HTTP Authorization server, you can add headers not included in the client request to the authorization request by defining the headers in the *Headers to add* parameter.
```
Comment (juanmolle) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]: For policies with an HTTP Authorization server, you can add headers not included in the client request to the authorization request by defining the headers in the *Headers to add* parameter.

I guess this shouldn't be added, in grpc server the server can add the headers too. probably we should just remove it

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 2]
```diff
@@ -16,11 +16,19 @@ endif::[]
...
+[IMPORTANT]
+====
+Flex Gateway send headers to the authorization service encoded in bytes. To send headers encoded in utf8, set the environment variable `FLEX_HEADER_RAW_VALUE_ENABLE=false`.
+====
+
+The External Authorization policy can also modify the client HTTP request and response headers with additional information returned from the authorization service. For example, if an authorized user makes a request with an access token header that corresponds to a username, the authorization service can return a `username` header for Flex Gateway to append to the client HTTP request.
+For gRPC authorization servers, if the authorization service returns headers with the authorization response, Flex Gateway either appends these headers to the client HTTP request or overwrites the header if it is present in the client request. For HTTP authorization servers, you can configure what headers are appended, overwritten, or ignored.
```
Comment (juanmolle) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]: well, related to previous comment, now say that with grpc can add/replace headers

**Review by juanmolle — APPROVED** [review]

**Review by juanmolle — APPROVED** [review]
LGTM

File: gateway/1.6/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 3]
```diff
@@ -20,7 +20,7 @@ The External Authorization policy forwards incoming client HTTP requests to an e
...
+When using a gRPC external authorization service, the External Authorization policy can also modify the client HTTP requests header with additional information returned from the authorization service. If the authorization service returns headers with the authorization response, Flex Gateway either appends these headers to the client HTTP request or overwrites the header if it is already present in the client HTTP request. For example, if an authorized user makes a request with an access token header that corresponds to a username, the authorization service can return a `username` header for Flex Gateway to append to the client HTTP request. When using the policy with an HTTP external authorization service, the policy cannot modify the client HTTP request.
```
Comment (Crispy-Salesforce) [inline gateway/1.6/modules/ROOT/pages/policies-included-external-authorization.adoc:23]: One of the sentence ("For example, if an authorized user...to append to the client HTTP request.") seems to be long. Did Vale flag it as a long sentence?

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 4]
```diff
@@ -16,11 +16,20 @@ endif::[]
...
+To forward request headers, such as an authorization token header, to the authorization service, use the *Allowed Headers* parameter to configure which headers from the client HTTP requests to send to the authorization service. For HTTP authorization servers, inject new headers into the authorization request by using the *Headers to add* parameter.
 
-The External Authorization policy can also modify the client HTTP requests header with additional information returned from the authorization service. If the authorization service returns headers with the authorization response, Flex Gateway either appends these headers to the client HTTP request or overwrites the header if it is already present in the client HTTP request. For example, if an authorized user makes a request with an access token header that corresponds to a username, the authorization service can return a `username` header for Flex Gateway to append to the client HTTP request.
+[IMPORTANT]
+====
+Flex Gateway send headers to the authorization service encoded in bytes. To send headers encoded in utf8, set the environment variable `FLEX_HEADER_RAW_VALUE_ENABLE=false`.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
Flex Gateway sends headers to the authorization service encoded in bytes. To send headers encoded in utf8, set the environment variable `FLEX_HEADER_RAW_VALUE_ENABLE=false`.
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 5]
```diff
@@ -16,11 +16,20 @@ endif::[]
...
+====
+
+The External Authorization policy can also modify the client HTTP request and response headers with additional information returned from the authorization service. For example, if an authorized user makes a request with an access token header that corresponds to a username, the authorization service can return a `username` header for Flex Gateway to append to the client HTTP request.
+
+For gRPC authorization servers, if the authorization service returns headers with the authorization response, Flex Gateway either appends these headers to the client HTTP request or overwrites the header if it is present in the client request. 
+
+For HTTP authorization servers, you can configure what headers are appended, overwritten, or ignored. Policy's with an HTTP server configured can also modify the client response returned from the upstream service.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
For HTTP authorization servers, you can configure what headers are appended, overwritten, or ignored. Policies with an HTTP server configured can also modify the client response returned from the upstream service. 
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 6]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
 | `false`
 | If `true`, includes peer certificates in the authorization server request. Supported values are `true` and `false`.
 
+| `serviceRequestHeadersToAdd`
+| Optional
+| NA
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]: (Global) If this is "Not applicable", it should read "N/A". See our MuleSoft CX Writing Style Reference, in the "N/A" entry (https://confluence.internal.salesforce.com/pages/viewpage.action?spaceKey=MTDT&title=MuleSoft+CX+Writing+Style+Reference)

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 7]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+| Defines custom headers and their value to inject into the authorization service request. This parameter is only available for `HTTP` type serves.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
| Defines custom headers and their value to inject into the authorization service request. This parameter is available for `HTTP` type serves.
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 8]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+
+| `serviceRequestHeadersToAdd.key`
+| Optional
+| NA
+| Name of the new header to add.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc:103]: (Global) Try to be consistent with the structure of the sentence in the table.
You can star with a verb "Defines custom headers..." or with a noun "Name of the..."

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 9]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+
+| `serviceRequestHeadersToAdd.value`
+| Optional
+| NA
+| Value of the new header to add.
+
+| `serviceResponseUpstreamHeaders`
+| Optional
+| NA
+| Specifies the headers from the authorization response to add to the client request to the upstream service. Flex Gateway overwrites coexisting headers. This parameter is only available for `HTTP` type serves.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
| Specifies the headers from the authorization response to add to the client request to the upstream service. Flex Gateway overwrites coexisting headers. This parameter is available for `HTTP` type serves.
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 10]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+
+| `serviceResponseUpstreamHeadersToAppend`
+| Optional
+| NA
+| Specifies the headers from the authorization response to add to the client request to the upstream service. Flex Gateway appends coexisting headers. This parameter is only available for `HTTP` type serves.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
| Specifies the headers from the authorization response to add to the client request to the upstream service. Flex Gateway appends coexisting headers. This parameter is available for `HTTP` type serves.
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 11]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+
+| `serviceResponseClientHeaders`
+| Optional
+| NA
+| Specifies the headers from the authorization response to add to the client response returned from the upstream service. Flex Gateway overwrites existing headers. This parameter is only available for `HTTP` type serves.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
| Specifies the headers from the authorization response to add to the client response returned from the upstream service. Flex Gateway overwrites existing headers. This parameter is available for `HTTP` type serves.
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 12]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+
+
+| `serviceResponseClientHeadersOnSuccess`
+| Optional
+| NA
+| Specifies the headers from the authorization response to add to the client response returned from the upstream service if the request is successfully authorized. Flex Gateway overwrites existing headers. This parameter is only available for `HTTP` type serves.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
| Specifies the headers from the authorization response to add to the client response returned from the upstream service if the request is successfully authorized. Flex Gateway overwrites existing headers. This parameter is available for `HTTP` type serves.
```

File: gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc [hunk 13]
```diff
@@ -75,10 +92,51 @@ In Local Mode, you apply the External Authorization policy to your API instance
...
+| `requestTimeout`
+| Optional
+| `5000`
+| Maximum time in milliseconds the policy waits for a response from the authentication service.
+
 | `allowedHeaders`
 | Optional
 | NA
-| By default, requests sent to gRPC servers include all client headers and requests sent to HTTP servers don't include any client headers. If you list allowed headers for either server type, requests include only the listed headers.
+| Specifies the headers from the client HTTP request to propagate to authorization request. By default, requests sent to gRPC servers include all client headers and requests sent to HTTP servers don't include any client headers. If you list allowed headers for either server type, requests include only the listed headers.
```
Comment (Crispy-Salesforce) [inline gateway/1.7/modules/ROOT/pages/policies-included-external-authorization.adoc]:
```suggestion
| Specifies the headers from the client HTTP request to propagate to authorization request. By default, requests sent to gRPC servers include all client headers. Requests sent to HTTP servers don't include any client headers. If you list allowed headers for either server type, requests include only the listed headers.
```

**Review by Crispy-Salesforce — APPROVED** [review]

**Review by Crispy-Salesforce — APPROVED** [review]

**Pull request merged**